src/Doctrine/DoctrineHelper.php

```diff
@@ -145,6 +146,19 @@ public function getMetadata(string $classOrNamespace = null, bool $disconnected
                 foreach ($loaded as $m) {
                     $cmf->setMetadataFor($m->getName(), $m);
                 }
+
+                // Invalidating the cached AnnotationDriver::$classNames to find new Entity classes
+                $metadataDriver = $em->getConfiguration()->getMetadataDriverImpl();
+                if ($metadataDriver instanceof MappingDriverChain) {
+                    foreach ($metadataDriver->getDrivers() as $driver) {
+                        if ($driver instanceof AnnotationDriver) {
```

Comment:
Wow! Yea, this issue is a bummer - but nice job at least finding a workaround.

Does this only affect the AnnotationDriver?

Reply:
And also for `StaticPHPDriver` but we only support annotation driver, right?